ExtractData/PKHeX.ExtractData/Utils/ExtractUtils.cs

```diff
@@ -0,0 +1,382 @@
+            {
+                Hasforms = Dexnumber_HasForms5_B1W1;
+                FormToDexNumber = FormToDexNumber5_B1W1;
+                FormNumberIndex = FormNumberIndex5_B1W1;
+            }
+
+            if (GData.Generation == 5 && !B1W1)
+            {
+                Hasforms = Dexnumber_HasForms5_B2W2;
+                FormToDexNumber = FormToDexNumber5_B2W2;
+                FormNumberIndex = FormNumberIndex5_B2W2;
+            }
+
+            if (Hasforms.Contains(indexnumber))
+            {
+                string[] formStrings = PKX.getFormList(indexnumber,
+                                         Util.getStringList("types", "en"),
+                                         Util.getStringList("forms", "en"), ExtractUtils.gendersymbols);
+                log += Environment.NewLine;
+                log += "FORM " + formStrings[0] + Environment.NewLine;
+            }
+
+            if (indexnumber > GData.MaxSpeciesGeneration)
+            {
+                if (indexnumber > GData.MaxSpeciesGeneration && FormToDexNumber[indexnumber - GData.MaxSpeciesGeneration - 1] > 0)
+                {
+                    int realdexnumber = FormToDexNumber[indexnumber - GData.MaxSpeciesGeneration - 1];
+                    log += "SPECIE " + species[realdexnumber] + " DEX NUMBER " + realdexnumber + Environment.NewLine;
+                    string[] formStrings = PKX.getFormList(realdexnumber,
+                                   Util.getStringList("types", "en"),
+                                   Util.getStringList("forms", "en"), ExtractUtils.gendersymbols);
+                    log += Environment.NewLine;
+                    log += "FORM " + formStrings[FormNumberIndex[indexnumber - GData.MaxSpeciesGeneration - 1]] + Environment.NewLine;
+                }
+                else  //494 495 in GEN 4
+                    log += "UNUSED DATA " + Environment.NewLine;
+            }
+
+            log += "INDEX NUMBER " + indexnumber + Environment.NewLine;
+            return log;
+        }
+
+        internal static byte[] PackTMHMTutorMoves(string Header, TMHMTutorMoves[] Moves,int Size)
+        {
+            using (var s = new MemoryStream())
+            using (var bw = new BinaryWriter(s))
+            {
+                bw.Write(Header.ToCharArray());
+                bw.Write((ushort)Moves.Length);
+                uint offset = (uint)(4 + (Moves.Length * 4) + 4);
+                for (int i = 0; i < Moves.Length; i++)
+                {
+                    bw.Write(offset);
+                    offset += (uint)Size;
+                }
+                bw.Write(offset);
+                for (int i = 0; i < Moves.Length; i++)
+                {
+                    bw.Write(Moves[i].setBits());
+                }
+                return s.ToArray();
+            }
+        }
+
+
+        internal static bool CheckEquals(TMHMTutorMoves[] Moves_Generated, TMHMTutorMoves[] Moves_Readed, out string diff)
```

Comment:
Yes, but i want to know exactly were is the first element where the data starting being different, because the da readed and generated should be the same, a different means an error in the process of writing and reading the data. That dont happens in the last version but it happened before it was completed